test/vstest.console.PlatformTests/AssemblyMetadataProviderTests.cs

```diff
@@ -0,0 +1,143 @@
+            if (framework.Equals("net451"))
+            {
+                // Reason is unknown for why full framework it is taking more time. Need to investigate.
+                expectedElapsedTime = 100;
+                Assert.AreEqual(actualFx.FullName, Constants.DotNetFramework451);
+            }
+            else if (framework.Equals("netcoreapp1.0"))
+            {
+                Assert.AreEqual(actualFx.FullName, Constants.DotNetFrameworkCore10);
+            }
+            else
+            {
+                Assert.AreEqual(actualFx.FullName, ".NETCoreApp,Version=v2.0");
+            }
+
+            Console.WriteLine("Framework:{0}, {1}", framework, string.Format(PerfAssertMessageFormat, expectedElapsedTime, stopWatch.ElapsedMilliseconds));
+            Assert.IsTrue(stopWatch.ElapsedMilliseconds < expectedElapsedTime, string.Format(PerfAssertMessageFormat, expectedElapsedTime, stopWatch.ElapsedMilliseconds));
+        }
+
+        [TestMethod]
+        public void GetFrameWorkForNativeDll()
+        {
+            var expectedElapsedTime = 5;
+            var assemblyPath = $@"{this.testEnvironment.PackageDirectory}\microsoft.testplatform.testasset.nativecpp\2.0.0\contentFiles\any\any\Microsoft.TestPlatform.TestAsset.NativeCPP.dll";
+            this.LoadAssemblyIntoMemory(assemblyPath);
+            var stopWatch = Stopwatch.StartNew();
+            var fx = this.assemblyMetadataProvider.GetFrameWork(assemblyPath);
+            stopWatch.Stop();
+            Assert.AreEqual(Framework.DefaultFramework.Name, fx.FullName);
+
+            Console.WriteLine(PerfAssertMessageFormat, expectedElapsedTime, stopWatch.ElapsedMilliseconds);
+            Assert.IsTrue(stopWatch.ElapsedMilliseconds < expectedElapsedTime, string.Format(PerfAssertMessageFormat, expectedElapsedTime, stopWatch.ElapsedMilliseconds));
+        }
+
+        private void TestDonetAssemblyArch(string projectName, string framework, Architecture expectedArch, long expectedElapsedTime)
+        {
+            var assemblyPath = this.testEnvironment.GetTestAsset(projectName + ".dll", framework);
+            this.LoadAssemblyIntoMemory(assemblyPath);
+            var stopWatch = Stopwatch.StartNew();
+            var arch = this.assemblyMetadataProvider.GetArchitecture(assemblyPath);
+            stopWatch.Stop();
+            Assert.AreEqual(expectedArch, arch, $"Expected: {expectedArch} Actual: {arch}");
+            Console.WriteLine("Framework:{0}, {1}", framework, string.Format(PerfAssertMessageFormat, expectedElapsedTime, stopWatch.ElapsedMilliseconds));
+            Assert.IsTrue(
+                stopWatch.ElapsedMilliseconds < expectedElapsedTime,
+                string.Format(PerfAssertMessageFormat, expectedElapsedTime, stopWatch.ElapsedMilliseconds));
+        }
+
+        private void LoadAssemblyIntoMemory(string assemblyPath)
+        {
+            // Load the file into RAM in ahead to avoid perf number(expectedElapsedTime) dependence on disk read time.
```

Comment:
I am not sure if this hypothesis is true: any file read once as data will not be read again.